### PR TITLE
Add httpx instrumenter

### PR DESCRIPTION
**Pull request by jforns**
- [x] Depends on https://github.com/socialpoint-labs/bsk/pull/51

### CODE REVIEW

File: httpx/instrument.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	status int
+}
+
+func (r *responseWriterDelegator) WriteHeader(code int) {
+	r.status = code
+	r.ResponseWriter.WriteHeader(code)
+}
+
+func httpStatusCodeClass(code int) string {
+	return strconv.Itoa(code/100) + "xx"
```
Comment (gonzaloserrano) [inline httpx/instrument.go]: what about `return fmt.Sprintf("%dxx", code/100)`?

File: httpx/instrument_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+func TestInstrument_RequestsDuration(t *testing.T) {
+	assert := assert.New(t)
+
+	for _, testCase := range []struct {
+		waitTime     time.Duration
+		deltaTime    time.Duration
+		tags         metrics.Tags
+		expectedTags int
+	}{
+		{time.Millisecond * 50, 2 * time.Millisecond * 50, nil, 4},
```
Comment (gonzaloserrano) [inline httpx/instrument_test.go]: suggestion: extract `time.Millisecond * 50` to a const

File: httpx/instrument_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+func TestInstrument_RequestsDuration(t *testing.T) {
+	assert := assert.New(t)
+
+	for _, testCase := range []struct {
+		waitTime     time.Duration
+		deltaTime    time.Duration
+		tags         metrics.Tags
+		expectedTags int
```
Comment (gonzaloserrano) [inline httpx/instrument_test.go]: could you add some headers to the test case? so it's obvius headers are not erased by the delegator

File: httpx/instrument.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/socialpoint-labs/bsk/metrics"
+)
+
+// InstrumentDecorator returns an adapter that instrument requests with some metrics:
+// - http.request_duration: requests duration
+// - http.requests: number of requests
+//
+// Metrics are tagged with the HTTP method, requests path, response status code and response status class.
+func InstrumentDecorator(met metrics.Metrics, t ...metrics.Tag) Decorator {
```
Comment (sp-daniel-ramos) [inline httpx/instrument.go]: There is already a function: InstrumentRequestDurationMetric

Comment (jforns) [reply httpx/instrument.go]: Right, but that's for the httpx client.

**Pull request merged**